File: app/src/main/java/com/example/blockgraming19/VariableExpression.java
package com.example.blockgraming19;

public final class VariableExpression implements Expression {
    
    private final String name;
    
    public VariableExpression(String name) {
        this.name = name;
    }

    @Override
    public Value eval() {
        if (!Variables.isExists(name)) throw new RuntimeException("Variable does not exists");
        return Variables.get(name);
    }

    @Override
    public String toString() {
        return name;
    }
}

File: app/src/main/java/com/example/blockgraming19/VariableExpression.kt
package com.example.blockgraming19

import com.example.blockgraming19.Variables
import java.lang.RuntimeException

class VariableExpression(private val name: String) : Expression {
    override fun eval(): Value {
        if (!Variables.isExists(name)) throw RuntimeException("Variable does not exists")
        return Variables.get(name)
    }

    override fun toString(): String {
        return name
    }
}